test/e2e/operator_expectations.go

```diff
@@ -340,3 +344,188 @@ MachineLoop:
+	if err := codec.DecodeProviderConfig(&capiv1alpha1.ProviderSpec{Value: dep.Spec.Template.Spec.ProviderConfig.Value}, &config); err != nil {
+		return err
+	}
+	if config.InstanceType == "m4.xlarge" {
+		config.InstanceType = "m4.large"
+	} else {
+		config.InstanceType = "m4.xlarge"
+	}
+	dep.Spec.Template.Spec.ProviderConfig = capiv1alpha1.ProviderSpec{
+		Value: &runtime.RawExtension{Object: &config},
+	}
+	err = wait.PollImmediate(1*time.Second, waitShort, func() (bool, error) {
+		if err := tc.client.Update(context.TODO(), &dep); err != nil {
+			glog.Errorf("error querying api for deployment object: %v, retrying...", err)
+			return false, nil
+		}
+		return true, nil
+	})
+	if err != nil {
+		return err
+	}
+	glog.Infof("New InstanceType for deployment %s is %s", dep.Name, config.InstanceType)
+
+	// Verify a new node per replica is rolled out
+	for i := 0; i < int(replicas); i++ {
+		glog.Infof("%d nodes remaining to finish rolling update", int(replicas)-i)
+		glog.Info("Verify a new node is rolled out")
+		glog.Info("Get new machine")
+		var newMachine capiv1alpha1.Machine
+		err = wait.PollImmediate(1*time.Second, waitLong, func() (bool, error) {
+			glog.Infof("Wait for a new machine to be created")
+			machineList := capiv1alpha1.MachineList{}
+			if err := tc.client.List(context.TODO(), &listOptions, &machineList); err != nil {
+				glog.Errorf("error querying api for machineList object: %v, retrying...", err)
+				return false, nil
+			}
+			glog.Infof("Initial number of machines: %d. current number of machines: %d", expectedNumberOfMachines, len(machineList.Items))
+			if expectedNumberOfMachines+1 != len(machineList.Items) {
+				return false, nil
+			}
+			sortedMachineList := machineList.Items
+			sort.Sort(MachineByCreationTimestamp(sortedMachineList))
+			newMachine = sortedMachineList[0]
+
+			return true, nil
+		})
+		glog.Info("Get new node")
+		err = wait.PollImmediate(1*time.Second, waitLong, func() (bool, error) {
+			glog.Infof("Wait for machine %s to have a nodeRef", newMachine.Name)
+			refreshedNewMachine := &capiv1alpha1.Machine{}
+			key := types.NamespacedName{
+				Namespace: namespace,
+				Name:      newMachine.Name,
+			}
+			if err := tc.client.Get(context.TODO(), key, refreshedNewMachine); err != nil {
+				glog.Errorf("error querying api for nodeList object: %v, retrying...", err)
+				return false, nil
+			}
+			if refreshedNewMachine.Status.NodeRef == nil {
+				return false, nil
+			}
+
+			newNodeRef := refreshedNewMachine.Status.NodeRef
+			newNode := &corev1.Node{}
+			key = types.NamespacedName{
+				Namespace: namespace,
+				Name:      newNodeRef.Name,
+			}
+			if err := tc.client.Get(context.TODO(), key, newNode); err != nil {
+				glog.Errorf("error querying api for nodeList object: %v, retrying...", err)
+				return false, nil
+			}
+			glog.Infof("Wait for new node %s to become available", newNode.Name)
+			return IsNodeReady(newNode), nil
+		})
+		if err != nil {
+			return err
+		}
+	}
```

Comment:
You test new machines are created. However, I don't see any test verifying new instances are of `m4.large` flavor.

Reply:
yes, we can add farther AWS API validation in follow ups once this settle

Reply:
But, is not the that whole point of the test? You run rolling update changing flavor from one to another? The deployment can flip the machineset but that does not guarantee the new machines have different flavors.

Reply:
Added a validation for the new machine to come with the expected type https://github.com/openshift/machine-api-operator/pull/164/files#diff-7a5de36d018010fe027cb621ddf222dfR759
Provider specific validations and iteration is wider topic that I'd like to scope separately